File: src/main/java/pw/react/backend/web/UserDto.java
package pw.react.backend.web;

import jakarta.validation.constraints.Email;
import pw.react.backend.models.UserEntity;

public record UserDto(Long id, String username, String password, @Email String email) {

    public static UserDto valueFrom(UserEntity user) {
        return new UserDto(user.getId(), user.getUsername(), null, user.getEmail());
    }

    public static UserEntity convertToUser(UserDto userDto) {
        UserEntity user = new UserEntity();
        user.setId(userDto.id());
        user.setUsername(userDto.username());
        user.setEmail(userDto.email());
        user.setPassword(userDto.password());
        return user;
    }
}

File: src/main/java/pw/react/backend/web/UserDto.kt
package pw.react.backend.web

import jakarta.validation.constraints.Email
import kotlinx.serialization.Serializable
import org.springframework.security.core.GrantedAuthority
import pw.react.backend.models.domain.User

@Serializable
data class UserDto(
    val id: Long? = null,
    val name: String,
    val lastName: String,
    val password: String?,
    val email: @Email String,
    val authority: String = "user"
) {

    fun toDomain() = User(name, lastName, email, password!!, id, authority)
}
